docs/070_Configuration/020_Spring_Integration.md

```diff
@@ -0,0 +1,210 @@
+obtain other data types which Hazelcast supports.
+
+ - multiMap
+ - replicatedmap
+ - queue
+ - topic
+ - set
+ - executorService
+ - idGenerator
+ - atomicLong
+ - atomicReference
+ - semaphore
+ - countDownLatch
+ - lock
+ 
+ ```xml
+<jet:hazelcast jet-instance-ref="jet-instance" id="hazelcast-instance"/>
+ 
+<hz:multiMap id="multiMap" instance-ref="hazelcast-instance" name="my-multiMap"/>
+
+<hz:replicatedMap id="replicatedMap" instance-ref="hazelcast-instance" name="my-replicatedMap"/>
+
+<hz:queue id="queue" instance-ref="hazelcast-instance" name="my-queue"/>
+
+<hz:topic id="topic" instance-ref="hazelcast-instance" name="my-topic"/>
+
+<hz:set id="set" instance-ref="hazelcast-instance" name="my-set"/>
+
+<hz:executorService id="executorService" instance-ref="hazelcast-instance" name="my-executorService"/>
+
+<hz:idGenerator id="idGenerator" instance-ref="hazelcast-instance" name="my-idGenerator"/>
+
+<hz:atomicLong id="atomicLong" instance-ref="hazelcast-instance" name="my-atomicLong"/>
+
+<hz:atomicReference id="atomicReference" instance-ref="hazelcast-instance" name="my-atomicReference"/>
+
+<hz:semaphore id="semaphore" instance-ref="hazelcast-instance" name="my-semaphore"/>
+
+<hz:countDownLatch id="countDownLatch" instance-ref="hazelcast-instance" name="my-countDownLatch"/>
+
+<hz:lock id="lock" instance-ref="hazelcast-instance" name="my-lock"/>
+```
+
+Hazelcast Jet also supports *lazy-init*, *scope* and *depends-on* bean attributes.
+
+```xml
+<jet:instance id="instance" lazy-init="true" scope="singleton">
+<!-- ... -->
+</jet:instance>
+<jet:client id="client" scope="prototype" depends-on="instance">
+<!-- ... -->
+</jet:client>
+```
+
+## Annotation-Based Configuration 
+
+Annotation-Based Configuration does not require any XML definition. Simply create 
+a class with related annotations, eg `@Configuration`. And provide JetInstance as a
+bean by annotating the method with `@Bean`. 
+
+```java
+@Configuration
+public class AppConfig {
+    
+    @Bean
+    public JetInstance instance() {
+        return Jet.newJetInstance();
+    }
+}
+``` 
+
+
+## Enabling SpringAware Objects
+
+You need to configure Hazelcast Jet with `<hz:spring-aware/>` tag or set 
+`SpringManagedContext` programmatically to enable spring-aware objects.
+Code samples for 
+[declarative](https://github.com/hazelcast/hazelcast-jet-code-samples/blob/master/spring/src/main/java/jet/spring/XmlConfigurationWithSchemaSample.java)
+and  
+[annotation-based](https://github.com/hazelcast/hazelcast-jet-code-samples/blob/master/spring/src/main/java/jet/spring/AnnotationBasedConfigurationSample.java)
+configurations are available at our Code Samples repo.
+
+You can mark your custom processors with `@SpringAware` annotation which 
+gives you the ability:
+
+ - to apply bean properties
+ - to apply factory callbacks such as `ApplicationContextAware`, `BeanNameAware`
+ - to apply bean post-processing annotations such as `InitializingBean`, `@PostConstruct`
+ 
```

Comment:
I'd prefer to first explain what is SpringAware, then how to configure it.

Also, I'd add an introduction. Roughly: Jet job is submitted to cluster, cluster has it's own Spring context. This is the way how to make the context accessible. 
Then the co